src/Foundation/NSUrlSessionHandler.cs

```diff
@@ -142,6 +145,15 @@ public NSUrlSessionHandler (NSUrlSessionConfiguration configuration)
 			if (configuration == null)
 				throw new ArgumentNullException (nameof (configuration));
 
+#if !MONOMAC
+			// if the configuration has an identifier, we are dealing with a background session, 
+			// therefore, we do not have to listen to the notifications.
+			if (string.IsNullOrEmpty (configuration.Identifier)) {
+				using (var notification = new NSString ("UIApplicationWillResignActiveNotification"))
+					notificationToken = NSNotificationCenter.DefaultCenter.AddObserver (notification, BackgoundNotificationCb);
```

Comment:
The observer is never removed, which means the BackgroundNotificationOn delegate contained in the observer will keep a permanent reference to the NSUrlSessionHandler instance, and it will never be collected by the garbage collector.

Reply:
that string constant is already a field/notification inside `UIApplication`, `WillResignActiveNotification`
any reason why you can't use it ?

Reply:
Yes, we are not compiling the class at a point in time in which we have access to it. Please take a look at the first commit build error in the bots :/

Reply:
That error looks like it was because it's compiling for watchOS where `UIApplication` is not available. 

https://developer.apple.com/documentation/uikit/uiapplicationwillresignactivenotification?language=objc does not mention watchOS has this constant.

Using the name manually you're _working_ around that - but it's unlikely the code works one watchOS (the code just make it looks like it should)

Reply:
I made that change locally and got the following compilation error:

`
System.Net.Http/HttpClientHandler.platformnotsupported.cs(129,63): warning CS1998: This async method lacks 'await' operators and will run synchronously. Consider using the 'await' operator to await non-blocking API calls, or 'await Task.Run(...)' to do CPU-bound work on a background thread.
HttpClientEx.cs(83,8): warning CS0219: The variable 'webExceptionStatus' is assigned but its value is never used
/Users/mandel/Xamarin/xamarin-macios/master/xamarin-macios/src/Foundation/NSUrlSessionHandler.cs(181,87): error CS0117: 'UIApplication' does not contain a definition for 'WillResignActiveNotification'
make[3]: *** [../../class/lib/monotouch_watch/reference/System.Net.Http.dll] Error 1
`

Nevertheless, I'll add the !WATCH to show that it is not done on WatchOS.

Reply:
the error just means it's not available on watchOS - your code avoid that (at compile time) but it won't change runtime (and you're also exposing a constant string that is not part of the public API on the platform which is, at best, risky).